mailbox.go

```diff
@@ -164,7 +165,10 @@ func newMailbox(s *Server, name, nsName string, size int) (*Mailbox, error) {
 				cancel()
 				return err != nil
 			})
-		if err != nil {
+		// Ingnore ErrNotOwner because most likely the previous owner panic'ed or exited badly. 
+		// So we'll ignore the error and let the mailbox creator retry later.  We don't want to panic
+		// in that case because it will take down more mailboxes and make it worse. 
+		if err != nil && err != registry.ErrNotOwner {
```

Comment:
We should check this earlier, otherwise we're doing unnecessary retries:

```go
	switch {
	case err == nil:
		return false
	// Ignore ErrNotOwner because most likely the previous owner panic'ed or exited badly. 
	// So we'll ignore the error and let the mailbox creator retry later.  We don't want to panic
	// in that case because it will take down more mailboxes and make it worse.
	case errors.Is(err, registry.ErrNotOwner):
		err = nil
		return false
	default:
		return true
	}
})
if err != nil {
	panic(fmt.Errorf("%w: unable to deregister mailbox: %v, error: %v", errDeregisteredFailed, nsName, err))
}
```